hibernate-core/src/main/java/org/hibernate/query/sqm/internal/ConcreteSqmSelectQueryPlan.java

```diff
@@ -318,6 +319,8 @@ private <T, X> T withCacheableSqmInterpretation(DomainQueryExecutionContext exec
 		CacheableSqmInterpretation localCopy = cacheableSqmInterpretation;
 		JdbcParameterBindings jdbcParameterBindings = null;
 
+		executionContext.getSession().autoPreFlush();
```

Comment:
We could scan the `domainParameterXref` in the constructor to decide if `autoPreFlush` is necessary, but let's first discuss this whole topic before actually implementing this.